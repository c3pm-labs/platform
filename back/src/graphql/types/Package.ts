import { objectType } from '@nexus/schema';

import { PackageService } from '../../services/PackageService';

export const Versions = objectType({
  name: 'Version',
  definition(t) {
    t.model.description();
    t.model.license();
    t.model.package();
    t.model.readme();
    t.model.publishedAt();
    t.model.version();
  },
});

export const Package = objectType({
  name: 'Package',
  definition(t) {
    t.model.author();
    t.model.name();
    t.model.versions();
    t.field('latest', {
      type: 'Version',
      resolve: async (pkg, _args, ctx) => {
        const packageService = new PackageService(ctx);
        return packageService.getLatestVersion(pkg.name);
      },
    });
  },
});
