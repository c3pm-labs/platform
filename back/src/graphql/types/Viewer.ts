import { objectType } from '@nexus/schema';

export const Viewer = objectType({
  name: 'Viewer',
  definition(t) {
    t.model('User').id();
    t.model('User').email();
    t.model('User').username();
  },
});
