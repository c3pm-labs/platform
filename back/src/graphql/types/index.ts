export * from './User';
export * from './Mutation';
export * from './Query';
export * from './Package';
