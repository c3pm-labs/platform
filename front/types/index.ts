export interface Package {
  name: string;
  latest: Version;
  versions: Version[];
  author: User;
  downloads: number;
  tags: string[];
}

export interface Version {
  version: string;
  publishedAt: string;
  description: string;
  readme: string;
  package?: Package;
}

export interface User {
  id: string;
  username: string;
  email: string;
  description: string;
  packages: Package[];
}

export interface Viewer extends User {
  username: string;
  email: string;
}
