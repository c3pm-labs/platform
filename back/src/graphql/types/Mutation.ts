import { mutationType, stringArg } from '@nexus/schema';

import { AuthService } from '../../services/AuthService';
import { UpdateService } from '../../services/UpdateService';

import { Viewer } from './Viewer';
import { User } from './User';

export const Mutation = mutationType({
  definition(t) {
    t.field('register', {
      type: Viewer,
      args: {
        username: stringArg({ required: true }),
        email: stringArg({ required: true }),
        password: stringArg({ required: true }),
      },
      resolve: async (parent, args, ctx) => {
        const authService = new AuthService(ctx);
        return authService.register(args);
      },
    });
    t.field('login', {
      type: Viewer,
      args: {
        login: stringArg({ required: true }),
        password: stringArg({ required: true }),
      },
      resolve: async (parent, args, ctx) => {
        const authService = new AuthService(ctx);
        return authService.login(args);
      },
    });
    t.field('logout', {
      type: Viewer,
      resolve: (parent, args, ctx) => {
        const authService = new AuthService(ctx);
        return authService.logout();
      },
    });
    t.field('update', {
      type: User,
      nullable: true,
      args: {
        id: stringArg(),
        username: stringArg(),
        email: stringArg(),
        description: stringArg(),
      },
      resolve: async (parent, args, ctx) => {
        const updateService = new UpdateService(ctx);
        return updateService.updateUser(args);
      },
    });
  },
});
