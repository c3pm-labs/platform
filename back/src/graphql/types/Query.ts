import { queryType, stringArg } from '@nexus/schema';

import { AuthService } from '../../services/AuthService';
import { UserService } from '../../services/UserService';
import { PackageService } from '../../services/PackageService';

import { Viewer } from './Viewer';
import { User } from './User';
import { Package, Versions } from './Package';

export const Query = queryType({
  definition(t) {
    t.field('viewer', {
      type: Viewer,
      resolve: async (parent, args, ctx) => {
        const authService = new AuthService(ctx);
        return authService.viewer();
      },
    });
    t.field('user', {
      type: User,
      nullable: true,
      args: {
        username: stringArg(),
        id: stringArg(),
      },
      resolve: async (parent, args, ctx) => {
        const userService = new UserService(ctx);
        return userService.findOne(args);
      },
    });
    t.list.field('search', {
      type: Package,
      args: {
        keyword: stringArg({ required: true }),
      },
      resolve: async (parent, args, ctx) => {
        const packageService = new PackageService(ctx);
        return packageService.search(args.keyword);
      },
    });
    t.field('package', {
      type: Package,
      args: {
        name: stringArg({ required: true }),
      },
      resolve: async (parent, args, ctx) => {
        const packageService = new PackageService(ctx);
        return packageService.getPackage(args.name);
      },
    });
    t.field('version', {
      type: Versions,
      args: {
        packageName: stringArg({ required: true }),
        version: stringArg({ required: false }),
      },
      resolve: async (parent, args, ctx) => {
        const packageService = new PackageService(ctx);
        return packageService.getVersionOrLatest(args.packageName, args.version);
      },
    });
  },
});
