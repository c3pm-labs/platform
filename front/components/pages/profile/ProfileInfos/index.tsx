import { Divider, makeStyles, Typography } from '@material-ui/core';
import { useRouter } from 'next/router';

import Button from '../../../Button';
import { User } from '../../../../types';
import { useViewer } from '../../../../hooks/auth';

export interface ProfileInfosProps {
  user: User;
}
const useStyles = makeStyles((theme) => ({
  name: {
    display: 'flex',
    marginTop: theme.spacing(3),
    fontSize: 22,
  },
  mail: {
    display: 'flex',
    marginTop: theme.spacing(2),
    fontSize: 18,
  },
  description: {
    display: 'flex',
    marginTop: theme.spacing(2),
    marginBottom: theme.spacing(2),
    fontSize: 16,
    color: theme.palette.grey[600],
  },
  divider: {
    width: '100%',
  },
  nbPackages: {
    display: 'flex',
    marginTop: theme.spacing(2),
    marginBottom: theme.spacing(5),
    fontSize: 18,
  },
}));

const ProfileInfos = ({ user }: ProfileInfosProps): JSX.Element => {
  const router = useRouter();
  const classes = useStyles();
  const CheckUser = (): boolean => {
    const viewer = useViewer();
    return viewer && user.email === viewer.email;
  };

  return (
    <>
      <Typography
        variant="subtitle1"
        className={classes.name}
      >
        {user ? user.username : 'error'}
      </Typography>
      <Typography
        variant="subtitle1"
        className={classes.mail}
      >
        {user ? user.email : 'error'}
      </Typography>
      <Typography
        variant="subtitle1"
        className={classes.description}
      >
        {user ? user.description : 'error'}
      </Typography>
      <Divider className={classes.divider} />
      <Typography
        variant="subtitle1"
        className={classes.nbPackages}
      >
        {!user || !user.packages ? '0' : user.packages.length}
        {' '}
        package
        {user && user.packages && user.packages.length > 1 ? 's ' : ' '}
        uploaded
      </Typography>
      {CheckUser()
        ? (
          <Button
            color="primary"
            variant="contained"
            type="submit"
            fullWidth
            onClick={(): void => {
              router.push('/settings');
            }}
          >
            Edit
          </Button>
        ) : null}

    </>
  );
};

export default ProfileInfos;
