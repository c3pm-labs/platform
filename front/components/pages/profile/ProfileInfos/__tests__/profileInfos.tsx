import React from 'react';
import { render } from '@testing-library/react';
import { User } from 'types';

import ProfileInfos from '../index';

// eslint-disable-next-line
const useViewer = jest.spyOn(require('hooks/auth'), 'useViewer');

test('ProfileInfos', () => {
  useViewer.mockImplementation(() => ({
    username: 'toto',
    email: 'toto@gmail.com',
  }));

  const { container } = render(
    <ProfileInfos
      user={
        {
          username: 'toto',
          email: 'toto@gmail.com',
          description: 'FulltStack',
          id: '12345',
        } as User
      }
      // eslint-disable-next-line no-console
      setIsEdit={() => console.log('toto')}
    />,
  );

  expect(container).toMatchSnapshot();
});

test('ProfileInfos no data', () => {
  useViewer.mockImplementation(() => null);

  const { container } = render(
    // eslint-disable-next-line no-console
    <ProfileInfos user={null} setIsEdit={() => console.log('toto')} />,
  );

  expect(container).toMatchSnapshot();
});
