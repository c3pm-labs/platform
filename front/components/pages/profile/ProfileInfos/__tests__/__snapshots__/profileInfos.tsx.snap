// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProfileInfos 1`] = `
<div>
  <h6
    class="MuiTypography-root makeStyles-name-1 MuiTypography-subtitle1"
  >
    toto
  </h6>
  <h6
    class="MuiTypography-root makeStyles-mail-2 MuiTypography-subtitle1"
  >
    toto@gmail.com
  </h6>
  <h6
    class="MuiTypography-root makeStyles-description-3 MuiTypography-subtitle1"
  >
    FulltStack
  </h6>
  <hr
    class="MuiDivider-root makeStyles-divider-4"
  />
  <h6
    class="MuiTypography-root makeStyles-nbPackages-5 MuiTypography-subtitle1"
  >
    0
     
    package
     
    uploaded
  </h6>
  <button
    class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-default-6 makeStyles-primary-7 makeStyles-primary-11 MuiButton-disableElevation MuiButton-fullWidth"
    tabindex="0"
    type="submit"
  >
    <span
      class="MuiButton-label"
    >
      Edit
    </span>
    <span
      class="MuiTouchRipple-root"
    />
  </button>
</div>
`;

exports[`ProfileInfos no data 1`] = `
<div>
  <h6
    class="MuiTypography-root makeStyles-name-15 MuiTypography-subtitle1"
  >
    error
  </h6>
  <h6
    class="MuiTypography-root makeStyles-mail-16 MuiTypography-subtitle1"
  >
    error
  </h6>
  <h6
    class="MuiTypography-root makeStyles-description-17 MuiTypography-subtitle1"
  >
    error
  </h6>
  <hr
    class="MuiDivider-root makeStyles-divider-18"
  />
  <h6
    class="MuiTypography-root makeStyles-nbPackages-19 MuiTypography-subtitle1"
  >
    0
     
    package
     
    uploaded
  </h6>
</div>
`;
